armi/physics/neutronics/tests/test_crossSectionManager.py

```diff
@@ -392,6 +395,22 @@ def test_addXsGroupsFromBlocks(self):
         self.assertEqual(len(blockCollectionsByXsGroup), 4)
         self.assertIn("AB", blockCollectionsByXsGroup)
 
+    def test_calcWeightedBurnup(self):
+        self.blockList[1].p.percentBu = 3.1
+        self.blockList[2].p.percentBu = 10.0
+        self.blockList[3].p.percentBu = 1.5
+        for b in self.blockList[4:]:
+            b.p.percentBu = 0.0
+        self.csm._updateBurnupGroups(self.blockList)
+        blockCollectionsByXsGroup = {}
+        blockCollectionsByXsGroup = self.csm._addXsGroupsFromBlocks(
+            blockCollectionsByXsGroup, self.blockList
+        )
+        self.assertEqual(
+            blockCollectionsByXsGroup["AA"]._calcWeightedBurnup(), 1 / 12.0
+        )
+        self.assertEqual(blockCollectionsByXsGroup["AB"]._calcWeightedBurnup(), 6.55)
```

Comment:
can you test AA too?

Reply:
I set a single block to 1.5 percent burnup and then zeroed out all of the others. There are 18 blocks in `AA`, so the average burnup is 1.5/18 = 1/12. That's what that assert is for.

Reply:
doh!